python/cudf/cudf/core/multiindex.py

```diff
@@ -1733,3 +1733,33 @@ def _intersection(self, other, sort=None):
         if sort is None and len(other):
             return midx.sort_values()
         return midx
+
+    def _split_columns_by_levels(self, levels):
+        # This function assumes that for levels with duplicate names, they are
+        # specified by indices, not name by ``levels``. E.g. [None, None] can
+        # only be specified by 0, 1, not "None".
+
+        # Normalize named levels into indices
+        if levels is not None:
+            level_indices = set()
+            level_names = list(self.names)
+            for lv in levels:
+                if isinstance(lv, int):
+                    level_indices.add(lv)
+                else:
+                    level_indices.add(level_names.index(lv))
+        else:
+            level_indices = range(len(self._data))
```

Comment:
Rather than splitting the function this way where you follow the same code path whether `level` is passed, I would recommend implementing a separate code path when levels is None. That code path can just put all the columns into the data_columns/column_names variables and return empty lists for the other two, skipping all the extra logic below.